### PR TITLE
Updates to latest actions and fixes golangci-lint drift 

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
I'll unmark as draft when windows works with lint again..

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,8 +10,6 @@ on:
 
 jobs:
   "CI":
-    env:
-      GO111MODULE: on
```
Comment (codefromthecrypt) [inline .github/workflows/ci.yml:14]: default for a while

File: .golangci.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 run:
```
Comment (codefromthecrypt) [inline .golangci.yml:1]: changes here are config drift and golint removed for revive

File: middleware/http/server_test.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ var (
 )
 
 func httpHandler(code int, headers http.Header, body *bytes.Buffer) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, _ *http.Request) {
```
Comment (codefromthecrypt) [inline middleware/http/server_test.go:34]: everything below here are new checks, but hopefully not a big deal

**Review by codefromthecrypt — COMMENTED** [review]
notes!

**Review by anuraaga — APPROVED** [review]

**Pull request merged**